lib/suppressions.cpp

```diff
@@ -373,26 +373,32 @@ bool SuppressionList::Suppression::parseComment(std::string comment, std::string
     return true;
 }
 
-bool SuppressionList::Suppression::isSuppressed(const SuppressionList::ErrorMessage &errmsg) const
+SuppressionList::Suppression::Result SuppressionList::Suppression::isSuppressed(const SuppressionList::ErrorMessage &errmsg) const
 {
-    if (hash > 0 && hash != errmsg.hash)
-        return false;
-    if (!errorId.empty() && !matchglob(errorId, errmsg.errorId))
-        return false;
     if (type == SuppressionList::Type::macro) {
         if (errmsg.macroNames.count(macroName) == 0)
-            return false;
+            return Result::None;
+        if (hash > 0 && hash != errmsg.hash)
+            return Result::Checked;
+        if (!errorId.empty() && !matchglob(errorId, errmsg.errorId))
+            return Result::Checked;
     } else {
         if (!fileName.empty() && !matchglob(fileName, errmsg.getFileName()))
-            return false;
+            return Result::None;
         if ((SuppressionList::Type::unique == type) && (lineNumber != NO_LINE) && (lineNumber != errmsg.lineNumber)) {
             if (!thisAndNextLine || lineNumber + 1 != errmsg.lineNumber)
-                return false;
+                return Result::None;
         }
+        if (hash > 0 && hash != errmsg.hash)
+            return Result::Checked;
+        // the empty check is a hack to allow wildcard suppressions on IDs to be marked as checked
+        if (!errorId.empty() && (errmsg.errorId.empty() || !matchglob(errorId, errmsg.errorId)))
+            return Result::Checked;
```

Comment:
I am not too happy with this (especially since empty IDs should not be allowed at all) but I did not see an other solution right now. We can take a look at that if the (unmatched) suppressions ever behave as expected and we actually have full test coverage. Still focused on at least being able to dogfood the `unmatchedSuppression` warning.

Reply:
We will have to see how this works out when everything is complete.